### PR TITLE
Rytherwy code review 1

### PR DESCRIPTION
**Pull request by wrytherUC**
#### Analysis of the program:
This project allows someone to find a park near them, allow them to search for parks, and provide details about parks that are found. Based on the design doc, it will allow users to see the weather at the found parks and see how active the parks are in the future. 

#### Was the program available in UC GitHub on time?
Yes, the program was available on time.

#### Is the program easy for you, as an outsider, to understand?
The program is easy to understand for a new user.

#### Does the program compile?
Yes, the program complies with no issues.

#### Three commits you made to your group's repository during Sprint 1: 
- Created the service interface : https://github.com/wrytherUC/TrailAnywhere/commit/2c5051ee909e1eab2e3c2b031582da36f16e714f
- Set up CircleCI for our repository: https://github.com/wrytherUC/TrailAnywhere/commit/a4d8dbdc396960de7e0c99c5ffbde78a1cf8abf1
- Add unit test for saving trails and adjusted code for the test to pass: https://github.com/wrytherUC/TrailAnywhere/commit/bdacb5a4f56ba76d7514401786234a23b1ea5c76

#### Three technical concepts I learned from this code review:
I did not learn anything new, but reinforced things we have been learning in this class. JavaDoc is important to have for documentation purposes. It makes it easier for any developer to look at the javadoc and connect the dots on what a class or function might be doing. DRY principles is good to have for reducing code duplication. The last items were enforcing the use of DTOs, DAOs, and persistence/repositories. DTOs are good for moving data between classes, DAOs are used for CRUD operations, and persistence is used to store data from DAOs.

### CODE REVIEW

File: src/main/java/com/npsproject/it4045npsproject/It4045NpsProjectApplication.java [hunk 1]
```diff
@@ -3,6 +3,9 @@
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 
+/**
+ * Starts the It4045NpsProject application
+ */
 @SpringBootApplication
 public class It4045NpsProjectApplication {
```
Comment (wrytherUC) [inline src/main/java/com/npsproject/it4045npsproject/It4045NpsProjectApplication.java:11]: Add more javadoc in increase readability and make it easier for other developers that might look at the code to understand. 
https://rhamedy.medium.com/a-short-summary-of-java-coding-best-practices-31283d0167d3
https://www.tutorialspoint.com/java/java_documentation.htm
https://blog.joda.org/2012/11/javadoc-coding-standards.html

File: src/main/java/com/npsproject/it4045npsproject/dto/ParkDetails.java
Comment (wrytherUC) [inline src/main/java/com/npsproject/it4045npsproject/dto/ParkDetails.java:1]: Removed a class that had the same attributes as the Park DTO. 
Keeping with DRY principles, any details about a park can be included with the Park DTO. 
https://www.digitalocean.com/community/tutorials/what-is-dry-development

File: src/main/java/com/npsproject/it4045npsproject/dto/ParkSearchRequest.java
Comment (wrytherUC) [inline src/main/java/com/npsproject/it4045npsproject/dto/ParkSearchRequest.java:1]: This should be a DAO. This class would be used to retrieve data from something like a database. 
https://kajalrawal.medium.com/dto-dao-and-mvc-concept-c460046b139b
https://wasteofserver.com/dto-dao-and-repository-patterns/

Comment (Fastovich) [reply src/main/java/com/npsproject/it4045npsproject/dto/ParkSearchRequest.java:1]: DAO is in there but named as parser

**Pull request closed without merging**